lib/server/monalisa-synchronization/MonALISAInterface.js

```diff
@@ -0,0 +1,169 @@
+            name: dataPass.name,
+            outputSize: dataPass.output_size ? Number(dataPass.output_size) : null,
+            reconstructedEventsCount: dataPass.reconstructed_events,
+            description: dataPass.description,
+            lastRunNumber: dataPass.last_run,
+        };
+    }
+
+    /**
+     * Process server response payload to database entites, remove unnecessary data
+     * @param {object<string, Partial<MonalisaDataPass>>} payload raw data acquired from MonALISA: mapping of Data Pass Name to its parameters
+     * @return {DataPass[]} adapted data
+     */
+    processDataPassesPayload(payload) {
+        return Object.entries(payload)
+            .map(([dataPassName, dataPassParamenters]) => ({
+                name: dataPassName.trim(),
+                ...dataPassParamenters,
+            }))
+            .filter((dataPass) => extractLhcPeriod(dataPass.name)?.year >= this.yearLowerLimit)
+            .map((dataPass) => this.monalisaDataPassToEntity(dataPass));
+    }
+
+    /**
+     * Get data pass details
+     * @param {string} description description property of a data pass
+     * @return {DataPassDetails} data pass details
+     */
+    async getDataPassDetails(description) {
+        const payload = await this.fetchDataPassDetails(description);
+
+        return this.processDataPassDetailsPayload(payload);
+    }
+
+    /**
+     * Fetch data pass properties from MonALISA
+     * @param {string} description description property of a data pass
+     * @return {object<number, { run_no: number }>} payload
+     */
+    async fetchDataPassDetails(description) {
+        const { default: fetch } = await import('node-fetch');
+        const specificURL = new URL(this.dataPassDetailsUrl);
+        specificURL.searchParams.set('res_path', 'json');
+        specificURL.searchParams.append('filter_jobtype', description);
+        return fetch(specificURL, this.httpOptions).then((response) => response.json());
+    }
+
+    /**
+     * Process server response payload to database entites, remove unnecessary data
+     * @param {object<number, { run_no: number }>} payload payload
+     * @return {DataPassDetails} adapted data
+     */
+    processDataPassDetailsPayload(payload) {
+        return { runNumbers: [...new Set(Object.values(payload).map(({ run_no }) => run_no))] };
+    }
+}
+
+exports.MonALISAInterface = MonALISAInterface;
+
+/**
+ * Create new instace of DataPassesSynchronizer
+ * @return {DataPassesSynchronizer} instance configured with environmental varaibles
+ */
+exports.setupMonALISAInterface = async () => new MonALISAInterface({
+    dataPassesUrl: monalisaConfiguration.dataPassesUrl,
+    dataPassDetailsUrl: monalisaConfiguration.dataPassDetailsUrl,
+    yearLowerLimit: monalisaConfiguration.dataPassesYearLowerLimit,
+    pfxCertificateBytes: await fs.readFile(monalisaConfiguration.userCertificate.path),
+    certificatePassphrase: monalisaConfiguration.userCertificate.passphrase,
+});
```

Comment:
Please update docblocks

I would name it `getMonALISAClient` or `createMonALISAClient`, and I would pass configuration to it to not make the factory depending on the environment configuration.

Reply:
removed